metis-core/metis-core-rest/src/main/java/eu/europeana/metis/core/rest/DatasetController.java

```diff
@@ -277,4 +414,56 @@ public Dataset getByDatasetName(@RequestHeader("Authorization") String authoriza
         responseListWrapper.getListSize(), nextPage);
     return responseListWrapper;
   }
+
+  /**
+   * Get all available countries that can be used.
+   * <p>The list is retrieved based on an internal enum</p>
+   *
+   * @param authorization the String provided by an HTTP Authorization header <p> The expected input
+   * should follow the rule Bearer accessTokenHere </p>
+   * @return The list of countries that are serialized based on {@link eu.europeana.metis.core.common.CountrySerializer}
+   * @throws BadContentException if the authorization header is un-parsable or the user cannot be
+   * authenticated
+   */
+  @RequestMapping(value = RestEndpoints.DATASETS_COUNTRIES, method = RequestMethod.GET, produces = {
+      MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE})
+  @ResponseStatus(HttpStatus.OK)
+  @ResponseBody
+  public List<Country> getDatasetsCountries(@RequestHeader("Authorization") String authorization)
+      throws BadContentException {
+
+    MetisUser metisUser = authenticationClient
+        .getUserByAccessTokenInHeader(authorization);
+    if (metisUser == null) {
+      throw new BadContentException(CommonStringValues.WRONG_ACCESS_TOKEN);
+    }
```

Comment:
This block here occurs everywhere in every call. Is there a way to externalize this into a more generic solution? This may be outside the scope of this task.